eureka2-core/src/main/java/com/netflix/eureka2/transport/base/BaseMessageConnection.java

```diff
@@ -174,6 +177,11 @@ public Boolean call(Object message) {
             public void call(Object o) {
                 metrics.incrementIncomingMessageCounter(o.getClass(), 1);
             }
+        }).doOnTerminate(new Action0() {
+            @Override
+            public void call() {
+                shutdown();
```

Comment:
I have added this action, to handle remote endpoint disconnects. Handling of onComplete and onError is up to the client.
I will add however a test that check, that in case of remote server disconnect, we detect if immediately and close the channel.